src/rp2040/hardware_structs/include/hardware/structs/usb.h

```diff
@@ -112,36 +120,446 @@ static_assert(sizeof(usb_host_dpram_t) == USB_DPRAM_MAX, "");
 static_assert(offsetof(usb_host_dpram_t, epx_data) == 0x180, "");
 
 typedef struct {
+    _REG_(USB_ADDR_ENDP_OFFSET)
+    // Device address and endpoint control
+    // 0x000f0000 [16-19] : ENDPOINT (0): Device endpoint to send data to
+    // 0x0000007f [0-6]   : ADDRESS (0): In device mode, the address that the device should respond to
     io_rw_32 dev_addr_ctrl;
-    io_rw_32 int_ep_addr_ctrl[USB_HOST_INTERRUPT_ENDPOINTS];
+
+    _REG_(USB_ADDR_ENDP1_OFFSET)
+    io_rw_32 int_ep_addr_ctrl[15];
+
+    _REG_(USB_MAIN_CTRL_OFFSET)
+    // Main control register
+    // 0x80000000 [31]    : SIM_TIMING (0): Reduced timings for simulation
+    // 0x00000002 [1]     : HOST_NDEVICE (0): Device mode = 0, Host mode = 1
+    // 0x00000001 [0]     : CONTROLLER_EN (0): Enable controller
     io_rw_32 main_ctrl;
-    io_rw_32 sof_rw;
+
+    _REG_(USB_SOF_WR_OFFSET)
+    // Set the SOF (Start of Frame) frame number in the host controller
+    // 0x000007ff [0-10]  : COUNT (0)
+    io_wo_32 sof_wr;
```

Comment:
This has changed from `io_rw_32 sof_rw;` to `io_wo_32 sof_wr;` - is this like #582 where it's a typo that we actually need to preserve for backwards-compatibility?

Reply:
I.e. any external code that was previously using `sof_rw` (if such code exists :wink: ) will now break, because it's been renamed to `sof_wr`.

Reply:
oops didn't see this, will fix